docs/proposals/resource-aggregation-proxy/README.md

```diff
@@ -0,0 +1,455 @@
+
+
+
+Apparently, use the proxy approach will greatly simplify external component's access to kubernetes cluster, and hide multiple cluster's implementation details from them. It's what we propose.
+
+
+
+
+Why not use :
+
+- **[Aggregated Kubernetes API Endpoint](https://github.com/karmada-io/karmada/blob/master/docs/userguide/aggregated-api-endpoint.md)** This component is built for Unified authentication, the restful api client still need to enumerate all the member clusters to get the right information.
+- **[Caching](https://github.com/karmada-io/karmada/tree/master/docs/proposals/caching) (aka karmada-search)** This component can only support get/list data, it cannot watch or update data. 
+
+# Motivation
+
+Write a Resource Aggregation Proxy for Karmada. This proxy provide native kubernetes api for components outside kubernetes. When client requests (include read & write operation) proxy, proxy will figure out the correct downstream kubenetes api servers (including Karmada API server and member clusters), request all of the selected servers, then merge the responses into one single response, then return the result to the client. External components can treat Karmada managed kubernetes clusters as one single cluster.
+
+## Goals
+
+Allow kubernetes api clients (kubectl, client-go, and other client use kubernetes api) read write following resources:
+
+- workload: create/get/list/watch/update/patch/delete
+- node: get/list/watch/update/patch
+- pod: get/list/watch/update/patch/logs/exec
+
+## Non-Goals
+
+- read write following resources:
+  - service account token
+  - event
+
+- Provide a solution for unique resource names across all member clusters.
+
+# Proposal
+
+Add a Resource Aggregation Proxy. When client try to list pod by label, this proxy will aggregate pods information from all member clusters, then return to client as a single list. When client try to get workload resource, proxy will get the information from Karmada API server directly.
+
+## User Stories
+
+### Story 1: Service Discovery
+
+Not everyone use kubernetes Service for service discovery. We use a private service discovery architecture to accommodate both containerized application and traditional application deployed on physical machines. 
+
+![user story - service discovery](user story - service discovery.svg)
+
+Its main workflow can be summaried as following:
+
+1. list/watch pods.
+2. When pod's ip changed, update the corresponding application's routing metadata.
+3. Patch the pod, add a custom annotation to mark the pod as already processed. 
+
+The service discovery controller ignore pod with this annotation, to avoid redundant routing metadata change. Workload redeployment or other operation caused pod to be deleted/replaced will automatically remove the added annotation, and make the pod eligible for next processing. 
```

Comment:
```suggestion
The service discovery controller ignores pods with this annotation to avoid redundant routing metadata change. Workload redeployment or other operations that cause pods to be deleted/replaced will automatically remove the added annotation, and make the pods eligible for the next steps of processing. 
```